addon/services/infinity.js

```diff
@@ -35,21 +35,31 @@ let cacheInfinityCollection = (_cachedCollection, infinityModel, identifier, tim
   return _cachedCollection[identifier] = { [future_timestamp]: infinityModel };
 };
 
+const ALLOWED_KEYS = [
+  'perPage', 'perPageParam',
+  'startingPage', 'firstPage',
+  'totalPagesParam', 'countParam',
+  'infinityCache', 'filter',
+  'storeFindMethod', 'meta'
+];
+
 /**
  * @method stringifyObjectValues
  * @param Object options
  * @param String identifier
  * @return String
  */
 let stringifyObjectValues = (options, identifier = '') => {
-  return Object.keys(options).reduce((acc, key) => {
-    const value = options[key];
-    if (!!value && typeof value === 'object') {
-      return stringifyObjectValues(value, acc);
-    }
+  return Object.keys(options)
+    .filter((key) => ALLOWED_KEYS.indexOf(key) > -1 || typeof options[key] === 'string')
```

Comment:
What if value is number or boolean (e.g. `this.model.infinity('model-name', { foo: 1, bar: true })`)? We may want a list of acceptable types as well.

Reply:
good idea...fixed!